### DISCUSSION TITLE
v4.0.1

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#4685] [Announcements]
v4.0.1

VS Code v1.63.0

code-server has been rebased on upstream's newly open-sourced server
implementation (#4414).

### Changed

- Web socket compression has been made the default (when supported). This means
  the `--enable` flag will no longer take `permessage-deflate` as an option.
- The static endpoint can no longer reach outside code-server. However the
  vscode-remote-resource endpoint still can.
- OpenVSX has been made the default marketplace.
- The last opened folder/workspace is no longer stored separately in the
  settings file (we rely on the already-existing query object instead).

### Removed

- Extra extension directories have been removed. The `--extra-extensions-dir`
  and `--extra-builtin-extensions-dir` flags will no longer be accepted.
- The `--install-source` flag has been removed.

### Deprecated

- `--link` is now deprecated (#4562).

### Security

- We fixed a XSS vulnerability by escaping HTML from messages in the error page (#4430).

Cheers 🎉

## New Contributors
* @AbdullahBinJahed made their first contribution in https://github.com/coder/code-server/pull/4230
* @jlandowner made their first contribution in https://github.com/coder/code-server/pull/4244
* @toddewilliams made their first contribution in https://github.com/coder/code-server/pull/4281
* @tianjianchn made their first contribution in https://github.com/coder/code-server/pull/4293
* @ChenJiaH made their first contribution in https://github.com/coder/code-server/pull/4369
* @mrfoxie made their first contribution in https://github.com/coder/code-server/pull/4336
* @im-coder-lg made their first contribution in https://github.com/coder/code-server/pull/4419
* @mauri made their first contribution in https://github.com/coder/code-server/pull/4430
* @arHSM made their first contribution in https://github.com/coder/code-server/pull/4472
* @JounQin made their first contribution in https://github.com/coder/code-server/pull/4546
* @Leask made their first contribution in https://github.com/coder/code-server/pull/4605
* @huangyoukun made their first contribution in https://github.com/coder/code-server/pull/4611
* @L3P3 made their first contribution in https://github.com/coder/code-server/pull/4579
* @elsbrock made their first contribution in https://github.com/coder/code-server/pull/4651

**Full Changelog**: https://github.com/coder/code-server/compare/v3.12.0...v4.0.1
**Milestone**: https://github.com/coder/code-server/milestone/34

<hr /><em>This discussion was created from the release <a href='https://github.com/coder/code-server/releases/tag/v4.0.1'>v4.0.1</a>.</em>

**Comment by jsjoeio** [comment]
This will be out on homebrew once this is merged: https://github.com/Homebrew/homebrew-core/pull/92582

**Comment by ucalyptus2** [comment]
@jsjoeio can u help answer #4727

**Reply by lg-nineoneninethreenine** [reply]
I answered, hope it's easy.

**Comment by fuyanbo123** [comment]


**Reply by lg-nineoneninethreenine** [reply]
Idk, elaborate. #elucidation-needed
